app/controllers/projects_controller.rb

```diff
@@ -63,7 +71,14 @@ def index
                      current_power.samples
                    end
 
-        @projects = @samples.group(:project).count
+        @projects = if only_library || exclude_library
+                      @samples.group(:project).count
+                    else
+                      # This check is so that we still return projects without any samples in the
+                      # default case. Ex: Project listing used by the CLI.
+                      # TODO: Optimize perf if it is an issue.
+                      current_power.projects.map { |p| [p, Sample.where(project: p).count] }
```

Comment:
Can you consolidate the above `only_updatable` logic with this?

Do `current_power.updatable_projects.map { |p| [p, Sample.where(project: p).count] }` if `only_updatable` is true.

This way we return a consistent object.

Reply:
👍